app/models/name.rb

```diff
@@ -2555,29 +2561,75 @@ def save_with_log(log = nil, args = {})
     save
   end
 
-  ################################################################################
+  ##############################################################################
+  #
+  #  :section: Limits
+  #
+  ##############################################################################
+
+  # :string field size limits in characters, based on this algorithm:
+  # Theoretical max differences between (text_name + author) and the field
+  #   search_name: 4
+  #     Adding a " sp." at the end is the worst case.
+  #   sort_name:   21
+  #     It adds " {N" and " !" (5 chars) in front of subgenus and the epithet
+  #     that goes with it. There can be up to 4 infrageneric ranks (subgenus,
+  #     section, subsection, stirps, plus the space between name and author.
+  #     That adds up to 5*4 + 1 = 21.
+  #   display_name: 41
+  #     Adds the space between name and author; "**__" and "__**" (8 chars)
+  #     around every epithet, grouping genus and species. Infrageneric ranks
+  #     win, making as many as 5 separate bold epithets or epithet pairs.
+  #     That adds up to 8*5 + 1 = 41.
+
+  # Numbers are hard-coded (rather than calculated) to make it easier to copy
+  # them to migrations.
+  def self.author_limit
+    # max if text_name_limit == 100, and we want to keep sort_name_limit < 256
+    134
```

Comment:
Short answer:  Yes.  To be precise:
If I increase author to accommodate the specific case, keep text_name unchanged, and adjust the other limits (search_name, display_name, sort_name) to their theoretical maxima,  then
 display_name == 261.

Long answer:
I want to increase `author` because of the specific case.  
Keeping `text_name` constant, I have to increase `search_name`, `display_name`, and `sort_name`.  
Because of the specific case, `author` must be increased to at least 120.  
With text_name unchanged at 100, the theoretical maxima are 
search_name: 120 + 100 + 4 = 224
display_name: 120 + 100 + 41 = 261
sort_name: 120 + 100 + 21 = 241

display_name is necessarily over the default :string limit, so I don't care if it goes up a little more.  The next largest is sort_name, which I can keep to 255. Therefore I kept sort_name at 255, and worked backwards from there.

author: 255 - 100 - 21  = 134
search_name: 134 + 100 + 4 = 238
display_name: 134 + 100 + 41 = 275

**I need to fix `display_name_limit` and the migration. They should be 275, not 285.**